test/tests/app_tree.rb

```diff
@@ -0,0 +1,70 @@
+require_relative '../test'
+require 'brakeman/app_tree'
+require 'fileutils'
+
+class AppTreeTests < Minitest::Test
+  def temp_dir_and_file_from_path(relative_path)
+    Dir.mktmpdir do |dir|
+      file = File.join(dir, relative_path)
+      FileUtils.mkdir_p(file)
+      FileUtils.touch(file)
+      yield dir, file
+    end
+  end
+
+  def test_ruby_file_paths
+    temp_dir_and_file_from_path("test.rb") do |dir, file|
+      at = Brakeman::AppTree.new(dir)
+      assert_equal [file], at.ruby_file_paths.collect(&:absolute)
+    end
+  end
+
+  def test_ruby_file_paths_skip_vendor_false
+    temp_dir_and_file_from_path("vendor/test.rb") do |dir, file|
+      at = Brakeman::AppTree.new(dir, :skip_vendor => false)
+      assert_equal [file], at.ruby_file_paths.collect(&:absolute)
+    end
+  end
+
+  def test_ruby_file_paths_skip_vendor_true
+    temp_dir_and_file_from_path("vendor/test.rb") do |dir, file|
+      at = Brakeman::AppTree.new(dir, :skip_vendor => true)
+      assert_equal [], at.ruby_file_paths.collect(&:absolute)
+    end
+  end
+
+  def test_ruby_file_paths_skip_vendor_true_add_libs_path
+    temp_dir_and_file_from_path("vendor/bundle/gems/gem123/lib/test.rb") do |dir, file|
+      at = Brakeman::AppTree.new(dir, :skip_vendor => true, :additional_libs_path => ["vendor/bundle/gems/gem123/lib"])
+      assert_equal [file], at.ruby_file_paths.collect(&:absolute)
+    end
+  end
+
+  def test_ruby_file_paths_skip_vendor_true_add_engine_path
+    temp_dir_and_file_from_path("vendor/bundle/gems/gem123/test.rb") do |dir, file|
+      at = Brakeman::AppTree.new(dir, :skip_vendor => true, :engine_paths => ["vendor/bundle/gems"])
+      assert_equal [file], at.ruby_file_paths.collect(&:absolute)
+    end
+  end
+
+  def test_ruby_file_paths_skip_vendor_true_tests_in_engine_path_still_excluded
+    temp_dir_and_file_from_path("vendor/bundle/gems/gem123/test/test.rb") do |dir, file|
+      at = Brakeman::AppTree.new(dir, :skip_vendor => true, :engine_paths => ["vendor/bundler/gems"])
+      assert_equal [], at.ruby_file_paths.collect(&:absolute)
```

Comment:
This test demonstrates that the existing global EXCLUDED_PATHS are still honored.  Even if we have tests or logs or anything else in EXCLUDED_PATHS and they reside in explicitly requested paths such as engine or lib paths, we still exclude them.